libs/game/console.ts

```diff
@@ -15,27 +15,29 @@ namespace game.consoleOverlay {
         consoleStrings = [];
     }
 
-    export function setVisible(value: boolean) {
+    export function setVisible(value: boolean, col?: number) {
         if (value != !!consoleStrings)
             consoleStrings = value ? [] : undefined;
+        if (col !== undefined)
+            consoleColor = col;
     }
 
     function listener(priority: ConsolePriority, text: string) {
         if (!consoleStrings)
             return;
-        else if (consoleStrings.length < consoleLines)
-            consoleStrings.push(text);
-        else {
-            for (let i = 1; i < consoleStrings.length; ++i) {
-                consoleStrings[i - 1] = consoleStrings[i];
-            }
-            consoleStrings[consoleStrings.length - 1] = text;
+
+        text.split("\n")
```

Comment:
is fast to do array.concat?

Reply:
concat returns a new array, so it would have to push all the old strings too instead of just the new ones https://github.com/microsoft/pxt/blob/master/libs/pxt-common/pxt-helpers.ts#L193